File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import "math"

func IsNumber(n int) bool {
	exponent := GetDigits(n)
	result := 0
	value := n
	for value > 0 {
		result += int(math.Pow(float64(value%10), float64(exponent)))
		value /= 10
	}
	return result == n
}

func GetDigits(n int) int {
	return int(math.Log10(float64(n))) + 1
}

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var digist = number.ToString().ToArray();
        var result = 0;
        foreach (var d in digist)
            result += (int)Math.Pow(d - '0', digist.Length);
        return result == number;
    }
}